client/sites-dashboard/components/test/sites-site-plan.tsx

```diff
@@ -0,0 +1,56 @@
+/**
+ * @jest-environment jsdom
+ */
+import { render } from '@testing-library/react';
+import { SitePlan } from '../sites-site-plan';
+
+const siteOwnerId = 1;
+const otherAdminId = 2;
+
+const stagingSite = {
+	ID: 1,
+	site_owner: siteOwnerId,
+	title: 'Staging Site',
+	is_wpcom_staging_site: true,
+};
```

Comment:
@p-jackson @Automattic/build Where might be a good place to put some standard mocks that conform to `SiteExcerptData` ?

Reply:
Controversial opinion: I'm in favour of copy/paste heavy code for tests. Unlike production code, I think it's good for tests to have basically zero abstractions so that it's super easy to see all the inputs and outputs. (I know, I know, `@testing-library` itself is an abstraction, but ignore that part 😅). I guess the downside is that we we ever added a new mandatory field to `SiteExcerptData` then there'd be lots of places to update. But given that's an easy change to make, compared to reading a test failure in code you're unfamiliar with, I'd make that trade off.

Anyway to answer the question, a pattern I'm familiar with in Calypso is the `client/state/selectors/test/fixtures/` folder. It's filled with objects that you can use to easily mock out a Redux store.

For `SiteExcerptData` I'd do something like `client/data/sites/test/fixtures/site-excerpt-data.ts`, putting the fixtures close to the hook that produces objects of that type. That way it'd be sitting alongside hypothetical tests in `client/data/sites/test/use-site-excerpts-query.ts`

Reply:
> Controversial opinion: I'm in favour of copy/paste heavy code for tests.

@p-jackson Fair enough! I'll leave it as-is for now.